package.json

```diff
@@ -1205,6 +1205,143 @@
           ],
           "scope": "window"
         },
+        "cmake.statusbar.type": {
```

Comment:
I would also prefer we name these `visibility` instead of `type` as I think that conveys the meaning better.

I need to do more testing (sorry, I haven't finished yet - it's been a busy week) but I was also thinking about the states because there is some inconsistency with how they appear when you use this top level setting vs. what I expected.  I will flesh this out more when I get a chance, but I did want to submit this review feedback in the meantime.

Reply:
Forgot to mention... I'm also considering alternate names for the enum values that work better with 'visibility' as the property name.  I'm open to feedback, but these are the names I'm thinking:
```
'default' or 'visible' or 'iconAndText'
'compact'
'icon' or 'iconOnly'
'hidden'
```

Reply:
changed it to `visiblity`.
enum values are now
`default`, `compact`, `icon` and `hidden`